packages/babel-parser/src/parser/lval.js

```diff
@@ -16,6 +16,7 @@ import type {
   /*:: ObjectMember, */
   /*:: TsNamedTypeElementBase, */
   /*:: Identifier, */
+  /*:: PrivateName, */
```

Comment:
Q (not related to this PR): why do we use flow comments here?

Reply:
Good question. These can be considered comment-only type imports used by the forward definitions such as
```js
+parseIdentifier: (liberal?: boolean) => Identifier;
  +parseMaybeAssignAllowIn: (
    refExpressionErrors?: ?ExpressionErrors,
    afterLeftParse?: Function,
    refNeedsArrowPos?: ?Pos,
  ) => Expression;
  +parseObjectLike: <T: ObjectPattern | ObjectExpression>(
    close: TokenType,
    isPattern: boolean,
    isRecord?: ?boolean,
    refExpressionErrors?: ?ExpressionErrors,
  ) => T;
```

because these methods are defined in derived class of `LValParser`. Frankly I would not oppose merging `LVal` with `Expression`, or even merge the whole `parser` folder into a single class so we don't have to deal with forward declarations.